pinot-common/src/test/java/org/apache/pinot/common/utils/request/RequestUtilsTest.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.utils.request;
+
+import org.apache.pinot.common.request.Expression;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class RequestUtilsTest {
```

Comment:
Suggest adding a query test. Currently it is not very clear why we need to do the conversion like this. Actually I don't know if always using double to strong float literal is correct

Reply:
I think we always use double but not float for data storage and processing?

I will add a query test. I have tested this and compared results with h2, this fix works (although I don't know if this works for all cases 😛 )